artemis-core-client/src/main/java/org/apache/activemq/artemis/core/message/impl/CoreMessage.java

```diff
@@ -622,6 +619,12 @@ private RuntimeException onCheckPropertiesError(Throwable e) {
    @Override
    public int getMemoryEstimate() {
       if (memoryEstimate == -1) {
+         // durable messages can enlarge buffer::capacity while persisted:
+         // better to encode it to properly account its max capacity
+         if (isDurable()) {
+            checkEncode();
```

Comment:
That means that a durable message would get encoded some time before then expected: @clebertsuconic it's something we can tolerate? 
if not the ideal thing would be to estimate it based on its state, but I'm not sure is feasable.